### PR TITLE
[MM-44031] add system console SMTP link to invitation error

### PR DESCRIPTION
**Pull request by salmanmanekia**
#### Summary
Add system console SMPT config link to the invitation error.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44031

#### Screenshots

|  before  |  after  |
|----|----|
| Screenshot 2022-10-08 at 16 45 15 | Screenshot 2022-10-08 at 16 22 29 |

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: actions/invite_actions.ts [hunk 1]
```diff
@@ -3,19 +3,22 @@
...
 import {addUsersToTeam} from 'actions/team_actions';
 import {t} from 'utils/i18n';
 import {localizeMessage} from 'utils/utils';
-import {isGuest} from 'mattermost-redux/utils/user_utils';
+import {isAdmin, isGuest} from 'mattermost-redux/utils/user_utils';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+
+import {Channel, ChannelMembership} from '@mattermost/types/channels';
+import {UserProfile} from '@mattermost/types/users';
+import {RelationOneToOne} from '@mattermost/types/utilities';
```
Comment (neallred) [inline actions/invite_actions.ts:20]: Nit: Import order should be:
* place the `@mattermost/types` imports above the mattermost-redux imports: it is an exposed package and therefore more external.
* new mattermost-redux imports should be grouped together with existing ones, at the top of the file.

**Review by neallred — CHANGES_REQUESTED** [review]
Looks like this works from a functionality perspective. We'll want to add internationalization support.

File: actions/invite_actions.ts (hunk 1 above)
Comment (salmanmanekia) [reply actions/invite_actions.ts:20]: hmm.. The IDE does the import arrangement for me on save. I have configured the IDE to use our .eslintrc.json on save but seems like its either missing some lint rules or using some extra rules. 
Screenshot 2022-10-12 at 15 47 42

File: components/invitation_modal/result_table.tsx [hunk 2]
```diff
@@ -138,6 +141,14 @@ export default function ResultTable(props: Props) {
                                     values={invitation.reason.values}
                                 />
                             );
+                        } else if (invitation.path && invitation.reason) {
+                            reason = (<>
+                                {invitation.reason}
```
[the remark this answers is not in the document]
Comment (salmanmanekia) [reply components/invitation_modal/result_table.tsx]: Thanks for the thorough explanation @neallred.

**Comment by salmanmanekia** [comment]
Thanks @neallred for a quick review.

**Comment by stevemudie** [comment]
@salmanmanekia Can you outline the steps I need to test this?

**Comment by salmanmanekia** [comment]
> @salmanmanekia Can you outline the steps I need to test this?

Hi @stevemudie, 

In the spinwick test server created for this PR you need to 'Invite a person' as a member and as a guest. The SMTP  setting in the test server has been configured in such a way that you should be seeing an error modal with details. In the details there is a link that once clicked should take the admin user to SMTP config page in admin console – this last thing is what has been done in this PR. I have also created a [loom](https://www.loom.com/share/d887fcd06205497c892952fecedbdda0) to demo this.

**Comment by stevemudie** [comment]
Thanks for the steps @salmanmanekia!

I wasn't encountering the link as expected before but did on this pass.

I just need to run E2E tests now then once that's done I should be able to approve it.

**Review by pvev — APPROVED** [review]
LGTM, thanks @salmanmanekia :shipit:

**Review by stevemudie — APPROVED** [review]

**Comment by annelieseklein** [comment]
What's SMTP?

**Review by annelieseklein — APPROVED** [review]

**Comment by salmanmanekia** [comment]
> What's SMTP?

SMTP is an internet communication protocol used for email transmission. In the system console we have an option for configuring it.

**Pull request merged**